### DISCUSSION TITLE
Horizontal color interlacing

### DISCUSSION THREAD

**Discussion by michal-barczewski** [Sakati84/3DPrintColorizer#52] [Ideas]
I'm having pretty bad issues with delamination on thin prints that don't have much or any infill to hold them together.

An idea to improve layer adhesion in this case:
change interlacing to also work in horizontal direction
what i mean is to color say 2 mm then don't color, then color 2mm again, and on next layer same but offset
This way there's a place for layers to stick together

like this:
```
---***---***---
***---***---***
---***---***---
```
And then inner wall could also be offset by one position so there's color behind non-colored area (this of course only makes sense for transparent filament), altho i'm not sure if it'd be accurate enough to not cover both anyway.

**Comment by Sakati84** [comment]
I started working on this a little but it's quite complicated to get this right.
A Problem I see is that it may be possible to see the pattern afterwards. I also think this needs to be solved in 2d and not in 1d. So not color 2mm but color a 2x2mm square. Since the lines can start at random positions at different positions they won't match otherwise.

**Reply by michal-barczewski** [reply]
It might be simpler to do this on model level - if it's possible to make my other idea with colors work :)
